src/bfd-disasm.cpp

```diff
@@ -114,7 +89,7 @@ AlignState BfdDisasm::is_aligned(uint64_t offset, uint64_t pc)
   uint64_t size = std::min(pc + 100, size_);
   void *buf;
 
-  if (fd_ < 0)
+  if (!std_filesystem::exists(fs_))
```

Comment:
If we're just going to open the fd later again I think it's better we use `open()` to avoid TOCTOU issues.

Reply:
@danobi replaced with buffer as a unique pointer, and `ifstream` instead of C-style open, pread

Reply:
I do not believe that solves the problem

Reply:
@danobi Hi, revered to FD